src/python/WMCore/Services/ReqMgrAux/ReqMgrAux.py

```diff
@@ -410,8 +410,20 @@ def listDiskUsageOverThreshold(config, updateDB):
 
     if updateDB and not t0Flag:
         agentDrainMode = bool(len(overThresholdDisks))
+        updateWMConfig = False
```

Comment:
Ahmed, if what we are trying to accomplish is to simply not set AgentDrainMode if UserDrainMode is already True, then I would suggest a potentially simpler variation for this. Something like:
```
agentDrainMode = bool(len(overThresholdDisks))
if agentConfig and (agentDrainMode != agentConfig["AgentDrainMode"]):
    if agentConfig["UserDrainMode"] == 'True':
        print("Agent is already in UserDrainMode: True mode. No need to update.")
    else:
        reqMgrAux.updateWMAgentConfig(config.Agent.hostName, {"AgentDrainMode": agentDrainMode},
                                          inPlace=True)
```

Reply:
updated Thank you Alan